bucket_test.go

```diff
@@ -464,6 +464,466 @@ func TestBucket_Delete_NonExisting(t *testing.T) {
+				insertRandKeysValuesBucket(t, bk, rand.Intn(4096))
+				createBucketIfNotExist(t, tx, tc.dstBucketPath...)
+
+				return nil
+			}); err != nil {
+				t.Fatal(err)
+			}
+			db.MustCheck()
+
+			// act
+			var tmpFile string
+			if err := db.Update(func(tx *bolt.Tx) error {
+				srcBucket := retrieveParentBucket(t, tx, tc.srcBucketPath...)
+				dstBucket := retrieveChildBucket(t, tx, tc.dstBucketPath...)
+				bucketToMove := tc.srcBucketPath[len(tc.srcBucketPath)-1]
+
+				// dump bucketToMove to bbolt file for assertion
+				bk := srcBucket.Bucket([]byte(bucketToMove))
+				tmpFile = dumpBucketToFile(t, bk)
+
+				mErr := srcBucket.MoveBucket([]byte(bucketToMove), dstBucket)
+				require.Equal(t, tc.expErr, mErr)
+
+				return nil
+			}); err != nil {
+				t.Fatal(err)
+			}
+			db.MustCheck()
+
+			// assert
+			if err := db.View(func(tx *bolt.Tx) error {
+				srcBucket := retrieveParentBucket(t, tx, tc.srcBucketPath...)
+				bucketToMove := tc.srcBucketPath[len(tc.srcBucketPath)-1]
+
+				if bk := srcBucket.Bucket([]byte(bucketToMove)); bk != nil {
+					t.Fatalf("expected childBucket %v to be moved from srcBucket %v", bucketToMove, srcBucket)
+				}
+
+				dstBucket := retrieveChildBucket(t, tx, tc.dstBucketPath...)
+				if bk := dstBucket.Bucket([]byte(bucketToMove)); bk == nil {
+					t.Fatalf("expected childBucket %v to be child of dstBucket %v", bucketToMove, dstBucket)
+				}
+
+				childBucket := dstBucket.Bucket([]byte(bucketToMove))
+				if childBucket == nil {
+					t.Fatalf("expected subBucket %v to exist within dstBucket %v", bucketToMove, dstBucket)
+				}
+
+				bucketOnDisk, err := os.ReadFile(tmpFile)
+				if err != nil {
+					t.Fatalf("error reading tmp file %v", tmpFile)
+				}
+
+				require.Equal(t, bucketOnDisk, childBucket.Write())
+
+				return nil
+			}); err != nil {
+				t.Fatal(err)
+			}
+			db.MustCheck()
+		})
+	}
+}
+func createBucketIfNotExist(t testing.TB, tx *bolt.Tx, paths ...string) *bolt.Bucket {
+	t.Helper()
+
+	bk, err := tx.CreateBucketIfNotExists([]byte(paths[0]))
+	if err != nil {
+		t.Fatalf("error creating bucket %v: %v", paths[0], err)
+	}
+
+	for _, key := range paths[1:] {
+		bk, err = bk.CreateBucketIfNotExists([]byte(key))
+		if err != nil {
+			t.Fatalf("error creating bucket %v: %v", key, err)
+		}
+	}
+
+	return bk
+}
+
+func retrieveParentBucket(t testing.TB, tx *bolt.Tx, paths ...string) *bolt.Bucket {
+	paths = paths[:len(paths)-1]
+	return retrieveChildBucket(t, tx, paths...)
+}
+
+func retrieveChildBucket(t testing.TB, tx *bolt.Tx, paths ...string) *bolt.Bucket {
+	t.Helper()
+
+	var bk *bolt.Bucket = nil
+	for _, path := range paths {
+		if bk == nil {
+			bk = tx.Bucket([]byte(path))
+		} else {
+			bk = bk.Bucket([]byte(path))
+		}
+		if bk == nil {
+			t.Fatalf("error retrieving bucket %v within paths %v", path, strings.TrimSuffix(strings.Join(paths, "->"), "->"))
+		}
+	}
+	return bk
+}
+
+func dumpBucketToFile(t testing.TB, bk *bolt.Bucket) string {
```

Comment:
@ahrtr  you asked to dump the bucket into a file for verification after moving 

is this correct ? 

i had to wrap the `bucket.write()` into an exposed `bucket.Write()` to use here